packages/forklift-console-plugin/src/modules/Plans/views/create/PlanCreatePage.tsx

```diff
@@ -59,65 +75,68 @@ export const PlanCreatePage: React.FC<{ namespace: string }> = ({ namespace }) =
   });
   useSaveEffect(state, dispatch);
 
-  const steps = [
-    {
-      id: 'step-1',
-      name: 'Select source provider',
-      component: (
-        <SelectSourceProvider
-          projectName={projectName}
-          filterState={filterState}
-          filterDispatch={filterDispatch}
-          dispatch={dispatch}
-          state={state}
-          providers={providers}
-          selectedProvider={selectedProvider}
-        />
-      ),
-      enableNext: filterState?.selectedVMs?.length > 0,
-    },
-    {
-      id: 'step-2',
-      name: 'Create migration plan',
-      component: (
-        <ProvidersCreateVmMigrationPage
-          state={state}
-          dispatch={dispatch}
-          emptyContext={emptyContext}
-        />
-      ),
-      enableNext:
-        !emptyContext &&
-        !(
-          !!state?.flow?.apiError ||
-          Object.values(state?.validation || []).some((validation) => validation === 'error') ||
-          state?.validation?.planName === 'default'
-        ),
-      canJumpTo: filterState?.selectedVMs?.length > 0,
-      nextButtonText: 'Create migration plan',
-    },
-  ];
+  const anyValidationError = Object.values(state?.validation || []).some(
+    (validation) => validation === 'error',
+  );
```

Comment:
consider useMemo and extract to a util file

Reply:
Extracted to a util file.
Regarding `useMemo` - again since it's fetched from the big migration state, see my comment above.

Reply:
As mentioned in previous comment, `state` is changed a lot and as a result, the validations, so there is no point in adding useMemo here.

Reply:
other parts in this component are causing rerenders that aren't related to the state - so you will run this even if the state isn't changing.

Reply:
the useMemo should have only state as a dep